lib/src/main/java/com/dynatrace/metric/util/DimensionList.java

```diff
@@ -140,15 +140,39 @@ String serialize() {
     boolean firstIteration = true;
 
     for (Dimension dimension : dimensions) {
-      if (!firstIteration) {
-        builder.append(",");
-      } else {
-        firstIteration = false;
-      }
+      // if the dimension is not valid, don't add it to the serialized line
+      if (isDimensionValid(dimension)) {
+        if (!firstIteration) {
+          builder.append(",");
+        } else {
+          firstIteration = false;
+        }
 
-      builder.append(dimension.serialize());
+        builder.append(dimension.serialize());
+      }
     }
 
     return builder.toString();
   }
+
+  static boolean isDimensionValid(Dimension dimension) {
+    // Dimension key should never be empty, as the creation of a dimension list will drop all
+    // dimension with empty keys. DimensionLists will always be normalized upon creation. At the
+    // point where this method is used, all dimensions should be part of a dimension list, and
+    // therefore never contain null or empty dimension keys.
+    String key = dimension.getKey();
+    if (key == null || key.isEmpty()) {
+      logger.warning("dimension key is null or empty.");
```

Comment:
would we want to log a warning here or also just add a debug log? :thinking: 
as we're only logging the metric as name as debug would we'd not be able to correlate it anyway, right? :thinking:

Reply:
It's hard to correlate these logs anyway. I assume one would need some sort of ID to match this log to the other one, but I think the standard java logging lib that we are using here does not allow for that, and I would prefer to not rip out all the logging code and introduce a dependency (which would then become a transitive dependency in consumers). So, the correlation will have to be done "manually" by looking at the logs and hoping that there aren't two parallel serializations happening. To come back to your question: I think as soon as we are dropping something, we should warn the user about it. This log line here should never happen, but the one below might. The one below is also the one that users would be interested in, I think.